### DISCUSSION TITLE
glowroot changes serialVersionUID?

### DISCUSSION THREAD

**Discussion by rvjansen** [glowroot/glowroot#1233] [Q&A] [answered]
WAS 9.0.5.15 S/390X - IBM J9 JVM 1.8.0_361

we are trying to add glowroot to all development and acceptance/performance testing WAS clusters, but we hit a snag with an application that cannot be run. This is what we see:

```
Failure loading serialized project as resource com.foo.bar.rules.RuleAgent; local class incompatible: stream classdesc serialVersionUID = -7177859044606055993, local class serialVersionUID = -7603240408021320315
```

(I have slightly mishandled the class name because of, you know, serious official business).

We have verified various times now that adding the glowroot agents causes the problem and removing it stops the problem from occurring. 
 
Q1: is there something with added instrumentation that might throw the application off with deserializing the classes?

Q2: I have learned that these classes also can be produced from a desktop version of that application: can it be that glowroot catches something it thinks is somewhat off, but acceptable to the application?

And of course: can we 'quarantine' a class prefix to avoid this?

Many thanks in advance!

**Comment by Never-lab** [comment, marked as answer]
## Triage

Plausible interaction between **Java serialization** and **bytecode weaving**.

Glowroot does not edit an explicit `serialVersionUID` field, but weaving adds fields/methods to instrumented classes. For `Serializable` classes **without** an explicit UID, the computed serialVersionUID changes → deserialization fails with exactly your error.

**Mitigations:** explicit `serialVersionUID` on affected classes; narrow/disable instrumentation for `com.foo.bar.rules.*`; see also #634.

**Quarantine:** no single exclude property — use instrumentation config to disable matching pointcuts/plugins.

Full answer: https://github.com/glowroot/glowroot/discussions/1229

**Verdict:** close as Q&A. If a specific class is woven unintentionally, open a new bug with class name + repro.

---
*Community triage*
